src/rules/noUnnecessaryTypeAssertionRule.ts

```diff
@@ -61,6 +67,9 @@ class Walker extends Lint.AbstractWalker<void> {
     }
 
     private verifyCast(node: ts.TypeAssertion | ts.NonNullExpression | ts.AsExpression) {
+        if (isAssertionExpression(node) && this.options.ruleArguments.indexOf(node.type.getText()) !== -1) {
```

Comment:
please pass the `sourceFile` as parameter to `.getText()` to avoid unnecessary lookup of the SourceFile